plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -460,6 +462,36 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         }
     }
 
+    if (ctx->cloud_logging_base_url) {
+        /*
+         * An alternate base URL was specified in the config. To avoid the confusion of a user
+         * not knowing whether the trailing `/` should be present, check whether the user
+         * provided it and remove it if it is.
+         */
+        cloud_logging_base_url_size = flb_sds_len(ctx->cloud_logging_base_url);
+        if (FLB_SDS_HEADER(
+                ctx->cloud_logging_base_url
+            )->buf[cloud_logging_base_url_size-1] == '/') {
+            cloud_logging_base_url_size -= 1;
+        }
+        cloud_logging_base_url_str = flb_sds_create_size(cloud_logging_base_url_size);
+
+        /* Note: The size calculated from `flb_sds_len` does not include the null terminator character,
+         * `size` argument for `flb_sds_snprintf` needs to be the size including the null terminator.
```

Comment:
Many invocations of `flb_sds_snprintf` use the direct result of `flb_sds_alloc` as the size, which does not count the null terminator; in all cases, those invocations do not adjust by +1. ([example](https://github.com/fluent/fluent-bit/blob/68f1887f46547561213db2e1e89571b04d3b8594/plugins/out_es/es.c#L367))

Though the docs for the underlying `snprintf` are clear that the size should count the null terminator...

I'm not sure what I'm missing but it seems that either the docs are wrong and `size` should not count the null, or all of the existing invocations of `flb_sds_snprintf` are wrong and should be counting it 🤔